loan-events/src/agent.spec.js

```diff
@@ -0,0 +1,274 @@
+const mockContract = {};
+
+jest.mock('forta-agent', () => ({
+  ...jest.requireActual('forta-agent'),
+  getEthersProvider: jest.fn(),
+  ethers: {
+    ...jest.requireActual('ethers'),
+    Contract: jest.fn().mockReturnValue(mockContract),
+  },
+}));
+
+const {
+  Finding, FindingType, FindingSeverity, ethers, TransactionEvent,
+} = require('forta-agent');
+
+const { provideHandleTransaction, provideInitialize } = require('./agent');
+const {
+  getObjectsFromAbi,
+  getEventFromConfig,
+  createMockEventLogs,
+} = require('./test-utils');
+
+const utils = require('./utils');
+
+const config = require('../bot-config.json');
+
+// utility function specific for this test module
+// we are intentionally not using the Forta SDK function due to issues with
+// jest mocking the module and interfering with default function values
+function createTransactionEvent(txObject) {
+  const txEvent = new TransactionEvent(
+    null,
+    null,
+    null,
+    [],
+    [],
+    null,
+    txObject.logs,
+    null,
+  );
+  return txEvent;
+}
+
+// check the configuration file to verify the values
+describe('check bot configuration file', () => {
+  it('procotolName key required', () => {
+    const { protocolName } = config;
+    expect(typeof (protocolName)).toBe('string');
+    expect(protocolName).not.toBe('');
+  });
+
+  it('protocolAbbreviation key required', () => {
+    const { protocolAbbreviation } = config;
+    expect(typeof (protocolAbbreviation)).toBe('string');
+    expect(protocolAbbreviation).not.toBe('');
+  });
+
+  it('developerAbbreviation key required', () => {
+    const { developerAbbreviation } = config;
+    expect(typeof (developerAbbreviation)).toBe('string');
+    expect(developerAbbreviation).not.toBe('');
+  });
+
+  it('contracts key required', () => {
+    const { contracts } = config;
+    expect(typeof (contracts)).toBe('object');
+    expect(contracts).not.toBe({});
+  });
+
+  it('contracts key values must be valid', () => {
+    const { contracts } = config;
+    Object.keys(contracts).forEach((key) => {
+      const { address, abiFile, events } = contracts[key];
+
+      // only check that an address exists for the LendingPoolAddressesProvider entry
+      if (key === 'LendingPoolAddressesProvider') {
+        expect(address).not.toBe(undefined);
```

Comment:
Any interest in adding a check that the `address` value is a valid address?
```suggestion
        expect(address).not.toBe(undefined);
        
        // check that the address is a valid address
        expect(ethers.utils.isHexString(address, 20)).toBe(true);
```